calico/felix/test/test_config.py

```diff
@@ -101,6 +105,49 @@ def test_invalid_action(self):
                                      "Invalid field value"):
             config = Config("calico/felix/test/data/felix_invalid_action.cfg")
 
+    def test_invalid_etcd(self):
+        data = {
+            "felix_invalid_scheme.cfg": "Invalid protocol scheme",
+            "felix_missing_key.cfg": "Missing etcd key",
+            "felix_missing_cert.cfg": "Missing etcd certificate"
+        }
+
+        for filename in data:
+            log.debug("Test filename : %s", filename)
+            with self.assertRaisesRegexp(ConfigException,
+                                         data[filename]):
+                config = Config("calico/felix/test/data/%s" % filename)
+
+    def test_unreadable_etcd_key(self):
+        with mock.patch("os.path.isfile", autospec=True) as m_isfile, \
```

Comment:
1. Yes
2. This is the only UT failure at the moment; everything else is passing